docs/developer/decision-records/2022-03-02-performance-tests/README.md

```diff
@@ -0,0 +1,53 @@
+# Performance Testing with Gatling
+
+## Decision
+
+We use Gatling to perform functional and performance testing.
+
+Performance tests are run over a single iteration as part of CI pipeline, to serve as functional system tests. The Gatling report is published to document trends over time.
+
+We plan to expand the scale of Gatling tests in order to test load, concurrency and stress scenarios.
+
+We will perform load testing of the application in order to:
+
+- Ensure application performance degrades gracefully as concurrency increases.
+- Measure response times for the main queries and operations on a regular basis to detect performance degradation.
+- Provide reusable harness for EDC integrators to validate that their application meets SLAs.
+
+## Rationale
+
+Load/concurrency tests should be developed and regularly exercised, even when no significant load is expected on an application, as they can help identify logical or configuration errors, that could otherwise manifest themselves in hard to reproduce ways. Conversely, if performance does not significantly degrade even under high load, it can indicate that the provisioned resources might be scaled down in order to save costs.
+
+We preferred to use a tool allowing writing tests in Java, to reduce learning curves and allows reusing any Java code that might be written in the future to help use the client API (e.g. for authentication).
+
+We wanted to avoid having to write tests in one framework for functional testing and in another framework for performance testing. Valid performance tests should include detailed assertions, hence cover the scope of functional tests.
```

Comment:
```suggestion
We want to avoid having to write tests in one framework for functional testing and in another framework for performance testing. Valid performance tests should include detailed assertions, hence cover the scope of functional tests.
```